emails/views.py

```diff
@@ -617,17 +620,46 @@ def _sns_message(message_json):
         wrapped_text = relay_header_text + text_content
         message_body["Text"] = {"Charset": "UTF-8", "Data": wrapped_text}
 
-    to_address = user_profile.user.email
-    formatted_from_address = generate_relay_From(from_address, user_profile)
-    response = ses_relay_email(
-        formatted_from_address,
-        to_address,
-        subject,
-        message_body,
-        attachments,
-        mail,
-        address,
+    resender_headers_flag, _ = Flag.objects.get_or_create(
+        name=RESENDER_HEADERS_FLAG_NAME,
+        defaults={"note": "MPP-2117: Filter-friendly mail headers"},
     )
+    resender_headers = (
+        resender_headers_flag.is_active_for_user(user_profile.user)
+        or resender_headers_flag.everyone
+    )
+    if resender_headers:
+        true_address = user_profile.user.email
+        reply_address = reply_address_for_user(user_profile)
+        from_header = generate_from_header(from_address)
+        response = ses_relay_email(
+            from_header=from_header,
+            to_header=true_address,
+            subject_header=subject,
+            reply_to_header=reply_address,
+            resent_from_header=to_address,
+            resent_sender_header=reply_address,
+            from_address=reply_address,
+            to_address=true_address,
+            message_body=message_body,
+            attachments=attachments,
+            mail=mail,
+            address=address,
+        )
+    else:
```

Comment:
suggestion (non-blocking): `else` code-smell to refactor this logic out of `_sns_message` and into a separate function with a single `if` block with an early `return` statement.